lnrpc/lightning.proto

```diff
@@ -3750,10 +3750,23 @@ message Payment {
     string payment_request = 9;
 
     enum PaymentStatus {
-        UNKNOWN = 0;
+        // Deprecated.
+        UNKNOWN = 0 [deprecated = true];
+
+        // Payment has inflight HTLCs.
         IN_FLIGHT = 1;
+
+        // Payment is settled.
         SUCCEEDED = 2;
+
+        // Payment is failed.
         FAILED = 3;
+
+        // Payment is created and has not attempted any HTLCs.
+        INITIATED = 4;
```

Comment:
Maybe useful to add to the PR description why it is useful to have this state? Does it matter to the caller whether a payment has or has not yet attempt any htlcs? And why is it good to distinguish between `INITIATED` and an `IN_FLIGHT` payment that has at that time for a brief moment no in flight htlcs - if that's possible.

Reply:
cool updated.

Reply:
>A new status StatusInitiated is added. When payment is in this status, it means the payment hasn't made any HTLC attempts yet. It's added to provide finer control over the payment, for instance, a payment under this status can be safely failed and removed, while StatusInflight cannot.

Isn't it the case that in some cases, an in-flight payment can be failed and removed too? I am thinking of a point where there are momentarily no in-flight htlcs, maybe after a result came back, but no new path has been determined yet?

Still the question remains why it is useful to not only have this state internally, but also expose it on the rpc interface? In a way it looks like an implementation detail that applications might start relying on for one reason or the other making it hard to touch that code again.

Reply:
And more generally, what is the reason for creating this PR in the first place. Is there a follow-up that is going to make use of these changes?

Reply:
> Isn't it the case that in some cases, an in-flight payment can be failed and removed too?

hmm what is the case?

> I am thinking of a point where there are momentarily no in-flight htlcs, maybe after a result came back, but no new path has been determined yet?

I think we can move this discussion into the following PR. Basically we register the htlc before attempting the path, so if you see a payment is in `StatusInflight`, it means you have to limited actions. There are two cases to see "momentarily no in-flight htlcs", 1) the payment is newly created, and 2) all the inflight htlcs are failed. I'm still thinking about adding a status for case 2 like how I did previously, but its effect is best seen in the followup PR.

> Still the question remains why it is useful to not only have this state internally

what's the pros and cons here? my thought is we can have atomic RPCs for MPP now.

> And more generally, what is the reason for creating this PR in the first place. Is there a follow-up that is going to make use of these changes or just general clean up?

good question. Updated the PR re the followup.

Reply:
>what's the pros and cons here? my thought is we can have atomic RPCs for MPP now.

Can you elaborate on that? How does distinguishing between `initiated` and `in flight` facilitate atomicity in the context of MPP?

Reply:
so we can have different rpc endpoints to init, update, fail, and delete a payment.

Reply:
I don't think I understand what you mean. How are these endpoints supposed to be used exactly, or is there a (draft) issue/pr that adds these rpc endpoints?

Reply:
I think you can view them as an API over the database. In my mind we could have endpoints like
- `payment/create`, that inits a payment.
- `payment/delete`, that deletes a payment, which we already have.
- `payment/update`, that's managed automatically atm, by our payment lifecycle.